src/jit/lclvars.cpp

```diff
@@ -235,6 +235,52 @@ void Compiler::lvaInitTypeRef()
 
     lvaInitArgs(&varDscInfo);
 
+#if FEATURE_FASTTAILCALL
+
+    //-------------------------------------------------------------------------
+    // Calculate the argument register usage.
+    //
+    // This will later be used for fastTailCall determination
+    //-------------------------------------------------------------------------
+
+    unsigned argRegCount      = 0;
+    unsigned floatingRegCount = 0;
+    unsigned stackArgCount    = 0;
+    size_t   stackSize        = 0;
+    unsigned compArgCount     = info.compArgsCount;
+
+    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc) {
+        varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
+    };
```

Comment:
For amd64 I am relying on lvaInitUserArgs to correctly increment the float and int counter. That will mean when we get to this point floatingRegCount + argRegCount <= 4.